coding/text_file_reader.hpp

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <functional>
+#include <string>
+#include <vector>
+
+namespace text_file_reader
+{
+using Callback = std::function<void(std::vector<std::string> const &, std::vector<std::string> const &)>;
+void ReadLineByLine(std::string const & fileName, Callback && fn);
```

Comment:
This is not a text file reader. This is like a CSV file reader. The name is misleading.